src/Button/PrintButton/PrintButton.tsx

```diff
@@ -0,0 +1,240 @@
+import React, {
+  useState
+} from 'react';
+
+import { useMap } from '../../Hook/useMap';
+import {
+  downloadBlob,
+  queuePrint,
+  getJobStatus
+} from '@camptocamp/inkmap';
+import SimpleButton from '../SimpleButton/SimpleButton';
+import _isFinite from 'lodash/isFinite';
+
+import {toLonLat} from 'ol/proj';
+import ImageWMS from 'ol/source/ImageWMS';
+import OlLayer from 'ol/layer/Layer';
+import OlMap from 'ol/Map';
+import OSM from 'ol/source/OSM';
+import TileWMS from 'ol/source/TileWMS';
+import VectorSource from 'ol/source/Vector';
+import WMTS from 'ol/source/WMTS';
+import GeoJSON from 'ol/format/GeoJSON';
+import VectorLayer from 'ol/layer/Vector';
+
+import OpenLayersParser from 'geostyler-openlayers-parser';
+
+import MapUtil from '@terrestris/ol-util/dist/MapUtil/MapUtil';
+import Logger from '@terrestris/base-util/dist/Logger';
+
+import {
+  WmsLayer,
+  WmtsLayer,
+  OsmLayer,
+  GeoJsonLayer,
+  InkmapLayer,
+  InkmapPrintSpec
+} from './InkmapTypes';
+
+interface OwnProps {
+  onProgressChange?: (val: number) => void;
+  outputFileName?: string;
+  dpi?: number;
+  size?: InkmapPrintSpec['size'];
+}
+
+export type PrintButtonProps = OwnProps;
+
+const PrintButton: React.FC<PrintButtonProps> = ({
+  dpi = 120,
+  onProgressChange,
+  outputFileName = 'react-geo-image.png',
+  size = [400, 240, 'mm'],
+  ...passThroughProps
+}) => {
+
+  const [loading, setLoading] = useState<boolean>(false);
+  const map = useMap();
+
+  const mapOlLayerToInkmap = async (olLayer: OlLayer): Promise<InkmapLayer | null> => {
```

Comment:
This could go into an Util class to keep the React component cleaner. It might also be useful somewhere else.

Reply:
moved it to ol-util: https://github.com/terrestris/ol-util/pull/705